File: app/schemas/admin_analytics.py
"""Schema模块：admin_analytics。"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

AdminRange = Literal["1d", "7d", "30d"]


class DashboardMetrics(BaseModel):
    request_count: int = 0
    hit_rate: float = 0
    avg_latency_ms: float = 0
    document_count: int = 0


class DashboardTrendPoint(BaseModel):
    bucket: str
    request_count: int = 0
    hit_count: int = 0
    avg_latency_ms: float = 0


class DashboardBreakdownItem(BaseModel):
    key: str
    label: str
    value: int = 0


class KnowledgeHeatItem(BaseModel):
    knowledge_point_id: str
    title: str
    count: int = 0


class AdminDashboardResponse(BaseModel):
    range: AdminRange
    metrics: DashboardMetrics
    trends: list[DashboardTrendPoint] = Field(default_factory=list)
    result_breakdown: list[DashboardBreakdownItem] = Field(default_factory=list)
    knowledge_heat: list[KnowledgeHeatItem] = Field(default_factory=list)


class GraphNode(BaseModel):
    id: str
    label: str
    node_type: Literal["knowledge_point", "question"]
    weight: float = 1.0
    is_orphan: bool = False
    link_count: int = 0


class GraphLink(BaseModel):
    source: str
    target: str
    weight: float = 1.0


class KnowledgeGraphSummary(BaseModel):
    node_count: int = 0
    edge_count: int = 0
    knowledge_point_count: int = 0
    question_count: int = 0


class KnowledgeGraphResponse(BaseModel):
    range: AdminRange
    nodes: list[GraphNode] = Field(default_factory=list)
    links: list[GraphLink] = Field(default_factory=list)
    summary: KnowledgeGraphSummary

File: front/shared/types/admin-analytics.ts
export type AdminRange = '1d' | '7d' | '30d'

export interface DashboardMetrics {
  request_count: number
  hit_rate: number
  avg_latency_ms: number
  document_count: number
}

export interface DashboardTrendPoint {
  bucket: string
  request_count: number
  hit_count: number
  avg_latency_ms: number
}

export interface DashboardBreakdownItem {
  key: string
  label: string
  value: number
}

export interface KnowledgeHeatItem {
  knowledge_point_id: string
  title: string
  count: number
}

export interface AdminDashboardResponse {
  range: AdminRange
  metrics: DashboardMetrics
  trends: DashboardTrendPoint[]
  result_breakdown: DashboardBreakdownItem[]
  knowledge_heat: KnowledgeHeatItem[]
}

export interface GraphNode {
  id: string
  label: string
  node_type: 'knowledge_point' | 'question'
  weight: number
  is_orphan?: boolean
  link_count?: number
}

export interface GraphLink {
  source: string
  target: string
  weight: number
}

export interface KnowledgeGraphSummary {
  node_count: number
  edge_count: number
  knowledge_point_count: number
  question_count: number
}

export interface KnowledgeGraphResponse {
  range: AdminRange
  nodes: GraphNode[]
  links: GraphLink[]
  summary: KnowledgeGraphSummary
}
